pkg/policy/groups/actions.go

```diff
@@ -127,53 +182,92 @@ func DeleteDerivativeCNP(cnp *cilium_v2.CiliumNetworkPolicy) error {
 	return nil
 }
 
-func addDerivativeCNP(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy) error {
-
+// DeleteDerivativeCCNP if the given policy has a derivative constraint, the
+// given CCNP will be deleted from store and the cache.
+func DeleteDerivativeCCNP(ccnp *cilium_v2.CiliumNetworkPolicy) error {
 	scopedLog := log.WithFields(logrus.Fields{
-		logfields.CiliumNetworkPolicyName: cnp.ObjectMeta.Name,
-		logfields.K8sNamespace:            cnp.ObjectMeta.Namespace,
+		logfields.CiliumClusterwideNetworkPolicyName: ccnp.ObjectMeta.Name,
 	})
 
-	var derivativeCNP *cilium_v2.CiliumNetworkPolicy
-	var derivativeErr error
+	if !ccnp.RequiresDerivative() {
+		scopedLog.Debug("CCNP does not have derivative policies, skipped")
+		return nil
+	}
+
+	err := k8s.CiliumClient().CiliumV2().CiliumClusterwideNetworkPolicies().DeleteCollection(
+		context.TODO(),
+		v1.DeleteOptions{},
+		v1.ListOptions{LabelSelector: fmt.Sprintf("%s=%s", parentCNP, ccnp.ObjectMeta.UID)})
+	if err != nil {
+		return err
+	}
+
+	DeleteDerivativeFromCache(ccnp)
+	return nil
+}
+
+func addDerivativePolicy(ctx context.Context, cnp *cilium_v2.CiliumNetworkPolicy, clusterScoped bool) error {
+	var scopedLog *logrus.Entry
+	if clusterScoped {
+		scopedLog = log.WithFields(logrus.Fields{
+			logfields.CiliumClusterwideNetworkPolicyName: cnp.ObjectMeta.Name,
+		})
+	} else {
+		scopedLog = log.WithFields(logrus.Fields{
+			logfields.CiliumNetworkPolicyName: cnp.ObjectMeta.Name,
+			logfields.K8sNamespace:            cnp.ObjectMeta.Namespace,
+		})
+	}
+
+	var (
+		derivativePolicy   *cilium_v2.CiliumNetworkPolicy
+		derivativeErr, err error
+	)
 
 	// The maxNumberOfAttempts is to not hit the limits of cloud providers API.
 	// Also, the derivativeErr is never returned, if not the controller will
 	// hit this function and the cloud providers limit will be raised. This
 	// will cause a disaster, due all other policies will hit the limit as
 	// well.
-	// If the createDerivativeCNP() fails, a new all block rule will be inserted and
+	// If the createDerivative{CNP,CCNP}() fails, a new all block rule will be inserted and
```

Comment:
Seems like this change does not reflect the code below.

Reply:
Ah, I forgot to change it back when I reverted some changes.